File: src/ObjectManager/Object.Tes/FilePacks/Records/345-STAT.Static.cs
using OA.Core;
using System;

namespace OA.Tes.FilePacks.Records
{
    public class STATRecord : Record, IHaveEDID, IHaveMODL
    {
        public override string ToString() => $"STAT: {EDID.Value}";
        public STRVField EDID { get; set; } // ID
        public FILEField MODL { get; set; } // NIF model

        public override bool CreateField(UnityBinaryReader r, GameFormatId formatId, string type, uint dataSize)
        {
            if (formatId == GameFormatId.Tes3)
                switch (type)
                {
                    case "NAME": EDID = new STRVField(r, dataSize); return true;
                    case "MODL": MODL = new FILEField(r, dataSize); return true;
                    default: return false;
                }
            return false;
        }
    }
}

File: src.mac/ObjectManager/ObjectManager/Object.Tes/FilePacks/Records/345-STAT.Static.swift
//
//  STATRecord.swift
//  ObjectManager
//
//  Created by Sky Morey on 5/28/18.
//  Copyright © 2018 Sky Morey. All rights reserved.
//

public class STATRecord: Record, IHaveEDID, IHaveMODL {
    public override var description: String { return "STAT: \(EDID)" }
    public var EDID: STRVField = STRVField_empty // Editor ID
    public var MODL: MODLGroup? = nil // Model

    override func createField(_ r: BinaryReader, for format: GameFormatId, type: String, dataSize: Int) -> Bool {
        switch type {
        case "EDID",
             "NAME": EDID = r.readSTRV(dataSize)
        case "MODL": MODL = MODLGroup(r, dataSize)
        case "MODB": MODL!.MODBField(r, dataSize)
        case "MODT": MODL!.MODTField(r, dataSize)
        default: return false
        }
        return true
    }
}
